src/bio/bio_device.c

```diff
@@ -699,7 +699,12 @@ led_device_action(void *ctx, struct spdk_pci_device *pci_device)
 		return;
 
 	pci_dev_type = spdk_pci_device_get_type(pci_device);
-	if (strcmp(pci_dev_type, "vmd") != 0) {
+	if (pci_dev_type == NULL) {
+		D_ERROR("nil pci device type returned\n");
+		opts->status = -DER_MISC;
+	}
+
+	if (strncmp(pci_dev_type, BIO_DEV_TYPE_VMD, strlen(BIO_DEV_TYPE_VMD)) != 0) {
```

Comment:
Do we have something more solid than strlen to use as a max? Some maximum for the dev type? We try not to use strlen for the same reason. If not, strcmp was basically doing the same thing. Sorry for the churn.

Reply:
I couldn't find a maximum value constant in the SPDK code for the pci device type, hence I just defined a constant. In SPDK strlen is used. I will change it back in my next PR.